ttnn/cpp/ttnn/operations/matmul/device/kernels/dataflow/reader_bmm_tile_layout_in0_sender_receiver_padding_block_sharded.cpp

```diff
@@ -112,7 +126,11 @@ void kernel_main() {
 
     for (uint32_t b = 0; b < batch; ++b) {
         for (uint32_t block = 0; block < num_blocks; ++block) {
-            const uint32_t block_id = block / num_blocks_per_shard;
+            uint32_t block_id = block / num_blocks_per_shard;
+            if constexpr (fuse_op) { // If used fused op, make block_id conform to ordering of tensor slices from all gather
+                block_id = fused_op_receiver.align_to_slice_and_sync(block, sender_id);
+            }
```

Comment:
is this gonna work if we have multiple blocks within a shard? Ex, two blocks per shard, does it produce the correct block_id?

Reply:
if all gather tensor slice contains multiple block, and we have multiple blocks within a shard, then we don't need `extract_shard_sub_blocks` to read again?

Reply:
and should only mcast sender receive the blocks?

Reply:
> is this gonna work if we have multiple blocks within a shard? Ex, two blocks per shard, does it produce the correct block_id?

Yes this would work, as long as the number of blocks in a shard equal the number of blocks in a tensor slice, see my comment [here](https://github.com/tenstorrent/tt-metal/pull/11760#discussion_r1737637586). The "sync" part of "align_to_slice_and_sync" only happens when the core is processing the first block of the shard/tensor slice. The remaining blocks can be processed as normal as we are sure that they are available in L1.